common/cache/domainCache.go

```diff
@@ -162,12 +179,83 @@ func (c *domainCache) getDomain(key, id, name string, cache Cache) (*domainCache
 
 func (entry *domainCacheEntry) duplicate() *domainCacheEntry {
 	result := newDomainCacheEntry()
-	result.Info = entry.Info
-	result.Config = entry.Config
-	result.ReplicationConfig = entry.ReplicationConfig
+	result.info = entry.info
+	result.config = entry.config
+	result.replicationConfig = entry.replicationConfig
+	result.configVersion = entry.configVersion
+	result.failoverVersion = entry.failoverVersion
+	result.isGlobalDomain = entry.isGlobalDomain
 	return result
 }
 
 func (entry *domainCacheEntry) isExpired(now time.Time) bool {
 	return entry.expiry.IsZero() || now.After(entry.expiry)
 }
+
+func (entry *domainCacheEntry) GetInfo() *persistence.DomainInfo {
+	return entry.info
+}
+
+func (entry *domainCacheEntry) GetConfig() *persistence.DomainConfig {
+	return entry.config
+}
+
+func (entry *domainCacheEntry) GetReplicationConfig() *persistence.DomainReplicationConfig {
+	return entry.replicationConfig
+}
+
+func (entry *domainCacheEntry) GetConfigVersion() int64 {
+	return entry.configVersion
+}
+
+func (entry *domainCacheEntry) GetFailoverVersion() int64 {
+	return entry.failoverVersion
+}
+
+func (entry *domainCacheEntry) GetIsGlobalDomain() bool {
+	return entry.isGlobalDomain
+}
+
+func (entry *domainCacheEntry) IsDomainActive() bool {
+	if !entry.isGlobalDomain {
+		// domain is not a global domain, meaning domain is always "active" within each cluster
+		return true
+	}
+	return entry.clusterMetadata.GetCurrentClusterName() == entry.replicationConfig.ActiveClusterName
+}
+
+func (entry *domainCacheEntry) ShouldReplicateEvent() bool {
+	// frontend guarantee that the clusters always contains the active domain, so if the # of clusters is 1
+	// then we do not need to send out any events for replication
+	return entry.clusterMetadata.GetCurrentClusterName() == entry.replicationConfig.ActiveClusterName &&
+		entry.isGlobalDomain && len(entry.replicationConfig.Clusters) > 1
+}
+
+func (entry *domainCacheEntry) GetDomainNotActiveErr() *domainNotActiveErr {
```

Comment:
Let's define this error as an helper somewhere in common.

Reply:
the idea here is to generate the error if and only if the domain is not active; 
i can separate the generation of error, see below, in common package. 
```
newDomainNotActiveErr(entry.info.Name, entry.clusterMetadata.GetCurrentClusterName(), entry.replicationConfig.ActiveClusterName)
```